dev/src/overmind/actions.tsx

```diff
@@ -557,14 +573,14 @@ export const fetchUserData = async ({ state, actions }: Context): Promise<boolea
 
 /** Switches between teacher and student view. */
 export const changeView = async ({ state, effects }: Context, courseID: number): Promise<void> => {
-    const enrollment = state.enrollmentsByCourseID[courseID]
-    if (isStudent(enrollment)) {
+    const enrollmentStatus = state.enrollmentsByCourseID[courseID].getStatus()
```

Comment:
Just wondering if there is a reason to get the status and check that instead of just doing as before using the `isStudent` on the `enrollment` instead??

Maybe then you can reuse the `enrollment` variable to set the status as well?? Or maybe I'm missing something about typescript variables.

```tsx
    const enrollment = state.enrollmentsByCourseID[courseID]
    if (isStudent(enrollment)) {
        const status = await effects.grpcMan.getEnrollmentsByUser(state.self.getId(), [Enrollment.UserStatus.TEACHER])
        if (status.data?.getEnrollmentsList().find(enrollment => enrollment.getCourseid() == courseID)) {
            enrollment.setStatus(Enrollment.UserStatus.TEACHER)
        }
    }
    if (isTeacher(enrollment)) {
        enrollment.setStatus(Enrollment.UserStatus.STUDENT)
```

Reply:
Or maybe `enrollment` here should be changed to `enroll` to avoid confusion between the two enrollment variables:
```ts
        if (status.data?.getEnrollmentsList().find(enroll => enroll.getCourseid() == courseID)) {
```

Reply:
Not quite sure why, but doing it like you suggest above, and earlier, prevent the UI from rerendering after changing view once.

Reply:
ok. Resolving